discovery/parse.go

```diff
@@ -27,6 +27,19 @@ func NewApp(name string, labels map[string]string) (*App, error) {
 	}, nil
 }
 
+func ExtractPort(name string) (nameWithoutPort string, port string) {
```

Comment:
This should be called from `NewAppFromString`
Also needs associated tests please.